indico/modules/events/client/js/admin.js

```diff
@@ -0,0 +1,49 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2023 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import autoLinkerRulesUrl from 'indico-url:events.autolinker_rules';
+
+import React, {useEffect, useState} from 'react';
+import ReactDOM from 'react-dom';
+import {Loader} from 'semantic-ui-react';
+
+import {indicoAxios} from 'indico/utils/axios';
+
+import AutoLinkerConfig from './components/AutoLinkerConfig';
+import AutoLinkerTestDrive from './components/AutoLinkerTestDrive';
+
+function AutoLinker() {
+  const [rules, setRules] = useState(null);
+
+  // Fetch the rules once, at page load time.
+  // We will rely on user input, set through <AutoLinkerConfig />, from now on.
+  useEffect(() => {
+    (async () => {
+      const {
+        data: {rules: newRules},
+      } = await indicoAxios.get(autoLinkerRulesUrl());
+      setRules(newRules);
+    })();
+  }, []);
+
+  return rules === null ? (
+    <Loader />
+  ) : (
+    <>
+      <AutoLinkerConfig rules={rules || []} onChange={setRules} />
+      <AutoLinkerTestDrive rules={rules || []} />
+    </>
+  );
+}
```

Comment:
You can use useIndicoAxios here:

```suggestion
function AutoLinker() {
  const [_rules, setRules] = useState(null);
  const {data, loading} = useIndicoAxios(autoLinkerRulesUrl(), {
    camelize: true,
  });

  const rules = _rules || data?.rules;

  return loading ? (
    <Loader />
  ) : (
    <>
      <AutoLinkerConfig rules={rules} onChange={setRules} />
      <AutoLinkerTestDrive rules={rules} />
    </>
  );
}
```

Reply:
Good point, thanks!